---
title: "Problem Set 2 Problems 1-3"
author: "Malay"
date: "12/8/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.

```{r}
#Problem 1
data("airquality")
knitr::kable(head(airquality))
correlation <- data.frame(cor(airquality, use="complete.obs", method="pearson"))
library('ggpmisc')

ut <- upper.tri(correlation)
corr <- data.frame(
    row = rownames(correlation)[row(correlation)[ut]],
    column = rownames(correlation)[col(correlation)[ut]],
    cor = (correlation)[ut])
View(corr)
my.formula <- y~x
ggplot(airquality, aes(x=Ozone, y=Temp)) + geom_point(shape=1) +
  geom_smooth(method=lm, formula= my.formula) + stat_poly_eq(formula = my.formula,aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")),
                parse = TRUE) +        
   geom_point()

```

```{r}
#Problem 2

#ignore this code
# s <- read.fasta(file =
                #   ("p53_all.fas"), seqtype="AA")
#  is.SeqFastaAA(s[[1]])
#  sublist <- getName(s)
# print(sublist)
# for ( NP_000537.3 in sublist) { getFrag(s, begin = 1, end = 7)}
# f<-s[names(s) %in% sublist]
# print.by(f, "NP_000537.3")
 
#!!!Use this code
library("seqinr")
s <- read.fasta(file =
                  ("p53_all.fas"), seqtype="AA")
is.SeqFastaAA(s[[1]])
getSequence(s$NP_000537.3, as.string = TRUE)

 

```
```{r}
##Problem 3
library("seqinr")
s <- read.fasta(file =
                  ("p53_all.fas"), seqtype="AA")
is.SeqFastaAA(s[[1]])
getSequence(s$NP_000537.3, as.string = TRUE)
#sublist <- getName(s)
#print(sublist)
#for ( NP_000537.3 in sublist) { getFrag(s, begin = 1, end = 7)}
#f<-s[names(s) %in% sublist]


#BiocManager::install("Biostrings")
library("Biostrings")
library("seqinr")
library("stringr")
#import sequences and BLOSUM matrix
a <- read.fasta(file = ("ex_align.fas"), seqtype="AA")
is.SeqFastaAA(a[[1]])
x<- data.frame(BLOSUM62)
#Check that it returns the correct score for amino acid alignment
print(x["N","R"])
#Separate the 2 sequences 
a1 <- a$P1
a2 <- a$P2
a2<- str_replace(a2,"-","NA")
print(a2)
#create strings because thats what biostrings uses
a1string <- c2s(a1)
a2string <- c2s(a2)
print(a1string)
print(a2string)
print(a1string)
#Internet says they need to be AA strings for pairwise alignment
seq1 <- AAString(a1[[1]])
print(seq1)
seq2 <- AAString(a2[[1]])
print(seq2)

for (i in ncola1[]){
  seq1 <- AAString(i)
  #print(seq1)
  for (j in a2[]) {
  seq2 <- AAString(j)
  #print(seq2)
  
  
myAlign <- pairwiseAlignment(pattern = seq1, subject = seq2, substitutionMatrix = "BLOSUM62", gapOpening = 0, gapExtension = 0, scoreOnly = TRUE)
print(myAlign)


  }}
print(x)
print(a1string)
print(a2string)

```

